src/pybind/mgr/dashboard_v2/controllers/rbd.py

```diff
@@ -68,3 +84,29 @@ def get(self, pool_name):
         if status == ViewCache.VALUE_EXCEPTION:
             raise value
         return {'status': status, 'value': value}
+
+    def create(self, data):
+        if not self.rbd:
+            self.rbd = rbd.RBD()
+
+        # Get input values
+        size = data.get('size', 4 * 1024 ** 3)
+        obj_size = data.get('obj_size', 2 ** 22)
+        features = data.get('features', [])
+        stripe_unit = data.get('stripe_unit', None)
+        stripe_count = data.get('stripe_count', None)
+        data_pool = data.get('data_pool', None)
+
+        # Set order
+        order = None
+        if obj_size > 0:
+            order = int(round(math.log(float(obj_size), 2)))
+
+        # Set features
+        feature_bitmask = self._format_features(features) if len(features) > 0 \
+            else 61  # FIXME: hardcoded int
+
+        ioctx = mgr.rados.open_ioctx(data['pool_name'])
+        self.rbd.create(ioctx, data['name'], size, order=order, old_format=False,
```

Comment:
IMO, instead of returning empty responses, we should return responses that contains useful information that can be used to increase the front-end usability.

What about this?
`{
  success: true 
}`
and
`{
  success: false,
  message: "Image myrbd already exists in pool mypool",
}`

> Note: This example can be implemented by catching the `ImageExists` exception.

If the backend does not catch this kind of exceptions, the best that frontend can do is displaying a generic error message that results from the _"500 - Internal Server Error"_, without useful information to the user.

Reply:
Yes

Reply:
If rbd creation fails, it will raise an exception which generates a standard JSON error with a status code != 201, thus `{ success: true }` or `success: false` is not necessary here. 

But I agree that we should return something useful. What about serializing the new RBD?

Reply:
I think we should differentiate between general python exceptions, that may occur due to a bug, and RBD exceptions that are thrown by the `RBD.create` function.

For python exceptions, those are already handled by our infrastructure that outputs a JSON with the traceback and status code 500.
For RBD, we should always return status code 200, and a JSON message similar to what @ricardoasmarques is proposing, where we specify a `success` flag, which will be `true` if the image was created, and `false` otherwise. In case of an RBD exception there are two approaches to communicate the error to the frontend:

a) return a string message with information about the error;
b) return an integer code (available in all RBD exceptions)  that describes the error

I'm in favor of the second approach (b) because that way the frontend can actually generate a string message that can be translated into several languages.

Reply:
Great, using an integer code (approach b) looks good.

Reply:
We should make use of HTTP status codes: 

| Code | Meaning               |
|------|-----------------------|
| 201  | Created               |
| 400  | Validation Error      |
| 500  | Internal Server Error |

there is no need for any additional status Boolean in the response. 

Our JSON exception handler already provides a standard way to generate error responses and we should make use of it as much as possible. Instead of defining new ways to return non-successful replies. This way, we only have only one JSON structure for errors.

Reply:
@sebastian-philipp I don't think an RBD create error fits in one of those three status code.
If we get an error because we are trying to create an image that already exists, it's not a validation error, nor an internal server error.

Reply:
@rjfd I thought this would be a validation error, because the given image name is wrong, so status 400 - Bad Request?